lib/kernel-manager.coffee

```diff
@@ -98,7 +74,13 @@ class KernelManager
 
 
     getAllKernelSpecs: ->
-        return _.map @_kernelSpecs, 'spec'
+        # TODO(nikita): switch to async and jupyter-js-services API
+        # Note that using sync here is a temporary regression, because it forces
+        # a manual call to update-kernels by the user.
+        # This will be fixed in a later patch, that adds proper async back
+        kernelSpecs = zmq_services.getZMQGateway().getKernelSpecsSync()
+        _.forEach kernelSpecs, (obj, name) -> obj.spec.name = name
```

Comment:
Gateway kernel spawning works using the kernel name (not display_name), which is not part of the kernelspec.

This whole method gets rewritten in patch 2, where I address this in a more comprehensive way.